### PR TITLE
feat: Add quality gate for chart name alignment with codebase name(#191)

### LINKED ISSUE #191: Implement Quality Gate for Helm Chart Name Alignment with Codebase Name
Acceptance Criteria:
- A quality gate is implemented in the review process that checks for alignment between the Helm chart name and the codebase name.
- A similar quality gate is implemented in the build pipeline to ensure name alignment before the build proceeds.
- In cases of misalignment, the quality gates should reject the review/build with a clear error message indicating the discrepancy between the Helm chart name and the codebase name.
- Documentation is updated to reflect the importance of naming alignment and the existence of these quality gates.

### PR DESCRIPTION
**Pull request by alezander86**
## Description

This Pull Request introduces a new feature designed to enforce naming conventions between Helm charts and their corresponding codebase names within our CI/CD pipelines. By implementing quality gates at both the review and build stages, we ensure that the Helm chart name aligns with the codebase name. This alignment is crucial for maintaining consistency, improving traceability, and facilitating a more streamlined development process.

Fixes #191

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How Has This Been Tested?

Testing was carried out in the following manner:
Unit tests were added to validate the functionality of the quality gates.
Integration tests were conducted within the CI/CD pipeline to ensure that the gates correctly identify discrepancies between Helm chart names and codebase names.
Manual tests were performed to cover edge cases and ensure the error messaging is clear and actionable.

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation to reflect the introduction of the quality gates and the importance of naming alignment.
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Pull Request contains one commit. I squash my commits.

### CODE REVIEW

File: charts/pipelines-library/templates/tasks/check-helm-chart-name.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  steps:
+    - name: check-helm-chart-name
+      env:
+        - name: CODEBASE_NAME
+          value: $(params.CODEBASE_NAME)
+        - name: CHART_DIR
+          value: deploy-templates
+      image: mikefarah/yq:4.44.1
+      script: |
+        #!/bin/sh
```
Comment (SergK) [inline charts/pipelines-library/templates/tasks/check-helm-chart-name.yaml:19]: let's simplify and use simple alpine image:
```
chart_name=$(awk '/^name:/ {print $2}' Chart.yaml)

if [ "$chart_name" == "$CODEBASE_NAME" ]; then
    echo "The name in Chart.yaml matches the CODEBASE_NAME."
else
    echo "The name in Chart.yaml does not match the CODEBASE_NAME."
fi
```

File: charts/pipelines-library/templates/pipelines/helm/gerrit-build-default.yaml [hunk 2]
```diff
@@ -53,6 +53,21 @@ spec:
 
 {{- include "gerrit-build-start" . | nindent 4 }}
 
+    - name: check-chart-name
+      params:
+        - name: CODEBASE_NAME
+          value: $(params.CODEBASE_NAME)
+        - name: chart-dir
```
Comment (SergK) [inline charts/pipelines-library/templates/pipelines/helm/gerrit-build-default.yaml:60]: do we need CHART_DIR?

Comment (alezander86) [reply charts/pipelines-library/templates/pipelines/helm/gerrit-build-default.yaml]: yes, we take a variable from the pipeline and pass it to the task, the same approach as for helm-docs, helm-lint, helm-template

**Review by SergK — APPROVED** [review]

**Review by NikolayMarusenko — APPROVED** [review]
